be/src/storage/rowset/column_reader.cpp

```diff
@@ -521,12 +534,25 @@ StatusOr<std::unique_ptr<ColumnIterator>> ColumnReader::new_iterator() {
             num_fields -= 1;
             ASSIGN_OR_RETURN(null_iter, (*_sub_readers)[num_fields]->new_iterator());
         }
+
+        std::vector<uint8_t> access_flags;
+        std::vector<ColumnAccessPath*> child_paths(num_fields, nullptr);
+        if (path != nullptr && !path->children().empty()) {
+            access_flags.resize(num_fields, 0);
+            for (const auto& child : path->children()) {
+                child_paths[child->index()] = child.get();
+                access_flags[child->index()] = 1;
+            }
```

Comment:
prefer `struct names` rather than `id` to access sub fields, as the struct may be with various headers.

Reply:
Don't worry, I was send the subfields name to storage. ColumnReader/SegementReader doesn't handle subfield name, so I use offset now

Reply:
1. if schema change, the offset is also right?
2. as in the future, the flat map and flat json have various schemas in each segment, the offset does not work. for example, need {a, b}, segment_0 may have {c,b,a), and segment_1 may have {d,e}, offsets does not work, we should use name to identify sub fields.

Reply:
1. map/json don't need schema change, and schema change has processed offset for other column
2. schema change on struct should follow the previous process
3. you can use name in json/map, and offset isn't work for map/json, I was support name/index. 

BTW, meterilized column is the internal implementation of map/json, and it should not be exposed to FE or Query,